File: 1833-find-the-highest-altitude/1833-find-the-highest-altitude.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int ans = INT_MIN;
        int n = gain.size();
        int temp = 0;

        for(int i = 0; i < n; i++){
            // Update the current altitude
            temp += gain[i];
            // Update the answer to store the maximum altitude
            ans = max(ans, temp);
        }

        return ans< 0 ? 0 : ans;
    }
};

File: 1833-find-the-highest-altitude/1833-find-the-highest-altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int n = gain.length;

        int ans = 0;

        ans = Math.max(ans, gain[0]);
        for(int i = 1; i < n; i++){
            gain[i] += gain[i-1];
            ans = Math.max(gain[i],ans);
        }

        return ans;
    }
}